packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -96,6 +96,8 @@ export const SelectPopover = styled( Ariakit.SelectPopover )`
 	border-radius: 2px;
 	background: ${ COLORS.theme.background };
 	border: 1px solid ${ COLORS.theme.foreground };
+	z-index: 9999; // Ensure the popover is on top
+	position: absolute !important;
```

Comment:
| Before | After |
|---------|---------|
| <img width="199" alt="Screenshot 2024-06-07 at 12 04 22 p m" src="https://github.com/WordPress/gutenberg/assets/81248/dfc165f3-c946-465d-9496-abc969ed7c87">  | <img width="197" alt="Screenshot 2024-06-07 at 12 13 22 p m" src="https://github.com/WordPress/gutenberg/assets/81248/ee71b013-51fb-4e69-8d23-2b256c1b7492"> |

Reply:
I'm not sure about the `position: absolute` here, since Ariakit should already set it automatically